samples/Sentry.Samples.OpenTelemetry.AspNetCore/Program.cs

```diff
@@ -0,0 +1,57 @@
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Sentry;
+using Sentry.OpenTelemetry;
+using Sentry.Samples.OpenTelemetry.AspNetCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// OpenTelemetry Configuration
+// See https://opentelemetry.io/docs/instrumentation/net/getting-started/
+builder.Services.AddOpenTelemetry()
+    .WithTracing(tracerProviderBuilder =>
+        tracerProviderBuilder
+            .AddSource(DiagnosticsConfig.ActivitySource.Name)
+            .ConfigureResource(resource => resource.AddService(DiagnosticsConfig.ServiceName))
+            .AddAspNetCoreInstrumentation()
+            .AddHttpClientInstrumentation()
+            .AddProcessor<SentrySpanProcessor>());
+
+// Use the Sentry propagator to ensure sentry-trace and baggage headers are propagated correctly.
+OpenTelemetry.Sdk.SetDefaultTextMapPropagator(new SentryPropagator());
+
+// You can use other propagators via composition, if needed.
+// For example, if you need both Sentry and W3C Trace Context propagation, then you can do the following:
+
+// OpenTelemetry.Sdk.SetDefaultTextMapPropagator(new CompositeTextMapPropagator(new TextMapPropagator[]
+// {
+//     new TraceContextPropagator(),
+//     new SentryPropagator()
+//
+//     // But don't include this.  It's already part of SentryPropagator.
+//     // new BaggagePropagator()
+// }));
+
+builder.WebHost.UseSentry(options =>
+{
+    options.EnableTracing = true;
+    options.Instrumenter = Instrumenter.OpenTelemetry;
```

Comment:
I'll do some digging.

Reply:
I came up with one way to do this... 

### Usage

```csharp
builder.AddOpenTelemetryWithSentry(
    tracerProviderBuilder => tracerProviderBuilder
            .AddSource(DiagnosticsConfig.ActivitySource.Name)
            .ConfigureResource(resource => resource.AddService(DiagnosticsConfig.ServiceName))
            .AddAspNetCoreInstrumentation()
            .AddHttpClientInstrumentation(),
    sentryOptions =>
    {
        sentryOptions.Dsn = "https://b887218a80114d26a9b1a51c5f88e0b4@o447951.ingest.sentry.io/6601807";
        sentryOptions.TracesSampleRate = 1.0;
        sentryOptions.Debug = builder.Environment.IsDevelopment();
    }
);
```

### Explanation

Basically where you had to call two separate extension methods before (each taking an action):

```
builder.Services.AddOpenTelemetry().WithTracing(tracerProviderBuilder => ...);
builder.WebHost.UseSentry(options => ...);
```

... now you call one instead, that accepts two actions as parameters. 

_**Pros:**_
- You don't have to call`tracerProviderBuilder..AddSentry()` in the first action
- You don't have to call `options.UseOpenTelemetry()` in the second action
- You either get OpenTelemetry wired up to work with Sentry and Sentry wired up to work with OpenTelemetry or you don't get either... it's not possible to just wire up half of it.

**_Cons:_**
- Somewhat hard to read with Lambda functions... I'd probably recommend people do with using Local functions to prevent their eyes from glazing over when they look at it.

### Implementation

This gives a rough idea of how it would work... it would pretty hack and slash and so is tightly coupled to `SentryAspNetCoreOptions`. Overloads etc. would need to be added for this to work with the other host types.

```csharp
public static class WebApplicationBuilderExtensions
{
    public static OpenTelemetryBuilder? AddOpenTelemetryWithSentry(
        this  WebApplicationBuilder? builder,
        Action<TracerProviderBuilder> userTracerProviderBuilder,
        Action<SentryAspNetCoreOptions>? userSentryOptions,
        TextMapPropagator? defaultTextMapPropagator = null
        )
    {
        if (builder is null)
        {
            return null;
        }

        void SentryOptionsWithOpenTelemetryInstrumentation(SentryAspNetCoreOptions options)
        {
            userSentryOptions?.Invoke(options);
            options.Instrumenter = Instrumenter.OpenTelemetry;
            options.AddTransactionProcessor(new OpenTelemetryTransactionProcessor());
        }

        builder.WebHost.UseSentry(SentryOptionsWithOpenTelemetryInstrumentation);

        // Configure OpenTelemetry to use Sentry's propagator (or some composite that includes it)
        defaultTextMapPropagator ??= new SentryPropagator();
        Sdk.SetDefaultTextMapPropagator(defaultTextMapPropagator);

        void TracerProviderBuilderWithSentrySpanProcessor(TracerProviderBuilder b)
        {
            userTracerProviderBuilder.Invoke(b);
            b.AddProcessor<SentrySpanProcessor>();
        }

        var openTelemetryBuilder = builder.Services.AddOpenTelemetry().WithTracing(
            TracerProviderBuilderWithSentrySpanProcessor
            );

        // Return an OpenTelemetryBuilder for chaining and further configuration of OpenTelemetry
        return openTelemetryBuilder;
    }
}
```

Reply:
I think if we decide to do that, we should do it as a separate Issue though, since this Pull Request is already getting pretty crazy and I'd like to merge it in before expanding the scope any further.

Reply:
Created [Improve initialisation of Sentry's OpenTelemetry integration #2460](https://github.com/getsentry/sentry-dotnet/issues/2460) to track this.